include/exadg/structure/material/library/st_venant_kirchhoff.cpp

```diff
@@ -149,35 +155,40 @@ StVenantKirchhoff<dim, Number>::evaluate_stress(
 
   if(dim == 3)
   {
-    S[0][0] = f0 * E[0][0] + f1 * E[1][1] + f1 * E[2][2];
-    S[1][1] = f1 * E[0][0] + f0 * E[1][1] + f1 * E[2][2];
-    S[2][2] = f1 * E[0][0] + f1 * E[1][1] + f0 * E[2][2];
-    S[0][1] = f2 * (E[0][1] + E[1][0]);
-    S[1][2] = f2 * (E[1][2] + E[2][1]);
-    S[0][2] = f2 * (E[0][2] + E[2][0]);
-    S[1][0] = f2 * (E[0][1] + E[1][0]);
-    S[2][1] = f2 * (E[1][2] + E[2][1]);
-    S[2][0] = f2 * (E[0][2] + E[2][0]);
+    S[0][0] = f0 * strain_measure[0][0] + f1 * (strain_measure[1][1] + strain_measure[2][2]);
+    S[1][1] = f0 * strain_measure[1][1] + f1 * (strain_measure[0][0] + strain_measure[2][2]);
+    S[2][2] = f0 * strain_measure[2][2] + f1 * (strain_measure[0][0] + strain_measure[1][1]);
+    S[0][1] = f2 * (strain_measure[0][1] + strain_measure[1][0]);
+    S[1][2] = f2 * (strain_measure[1][2] + strain_measure[2][1]);
+    S[0][2] = f2 * (strain_measure[0][2] + strain_measure[2][0]);
+    S[1][0] = S[0][1];
+    S[2][1] = S[1][2];
+    S[2][0] = S[0][2];
   }
   else
   {
-    S[0][0] = f0 * E[0][0] + f1 * E[1][1];
-    S[1][1] = f1 * E[0][0] + f0 * E[1][1];
-    S[0][1] = f2 * (E[0][1] + E[1][0]);
-    S[1][0] = f2 * (E[0][1] + E[1][0]);
+    S[0][0] = f0 * strain_measure[0][0] + f1 * strain_measure[1][1];
+    S[1][1] = f1 * strain_measure[0][0] + f0 * strain_measure[1][1];
+    S[0][1] = f2 * (strain_measure[0][1] + strain_measure[1][0]);
+    S[1][0] = S[0][1];
   }
 
   return S;
 }
 
 template<int dim, typename Number>
 dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-StVenantKirchhoff<dim, Number>::apply_C(
-  dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & E,
+StVenantKirchhoff<dim, Number>::second_piola_kirchhoff_stress_derivative(
+  dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & Grad_delta,
+  dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & F_lin,
   unsigned int const                                              cell,
   unsigned int const                                              q) const
 {
-  return evaluate_stress(E, cell, q);
+  // Exploit linear stress-strain relationship and minor symmetries in C
+  // when forming directional derivative Du(PK2) = Du(C : E) = C : Du(E)
+  // = C : (0.5 * ( transpose(F_lin) * Grad_delta + transpose(Grad_delta) * F_lin ))
+  // = C : (transpose(F_lin) * Grad_delta )
+  return (this->second_piola_kirchhoff_stress(transpose(F_lin) * Grad_delta, cell, q));
```

Comment:
If we change the implementation according to comment https://github.com/exadg/exadg/pull/537/files#r1284082874, this line will probably no longer be correct, but we would need to pass in a symmetric tensor, right?